ecosystem/typescript/sdk_v2/src/crypto/multi_ed25519.ts

```diff
@@ -0,0 +1,159 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/* eslint-disable no-bitwise */
+import { Deserializer, Serializer } from "../bcs";
+import { Uint8 } from "../types";
+import { Ed25519PublicKey, Ed25519Signature } from "./ed25519";
+
+/**
+ * MultiEd25519 currently supports at most 32 signatures.
+ */
+const MAX_SIGNATURES_SUPPORTED = 32;
+
+export class MultiEd25519PublicKey {
+  /**
+   * Public key for a K-of-N multisig transaction. A K-of-N multisig transaction means that for such a
+   * transaction to be executed, at least K out of the N authorized signers have signed the transaction
+   * and passed the check conducted by the chain.
+   *
+   * @see {@link
+   * https://aptos.dev/guides/creating-a-signed-transaction#multisignature-transactions | Creating a Signed Transaction}
+   *
+   * @param public_keys A list of public keys
+   * @param threshold At least "threshold" signatures must be valid
+   */
+  constructor(public readonly public_keys: Ed25519PublicKey[], public readonly threshold: Uint8) {
```

Comment:
wonder if we just use `threshold : number`

Reply:
Since the max threshold is 32, I think Uint8 should be fine here?

Reply:
`Uint8` is just a number type https://github.com/aptos-labs/aptos-core/blob/main/ecosystem/typescript/sdk_v2/src/types/index.ts#L11 - we declare it as Uint8 for BCS purposes. Since all BCS related code can technically live in its own library and outside the SDK, we should avoid using BCS related types throughout the SDK (might be wiser to move BCS types under the `bcs` folder and not place them in the global sdk types file - will add it as an item to the roadmap).

Reply:
I see! Will update it to number.